ros2_batch_job/__main__.py

```diff
@@ -407,10 +416,14 @@ def run(args, build_function, blacklisted_package_names=None):
     job = None
 
     args.white_space_in = args.white_space_in or []
+    def add_white_space_to_path(path, space_name):
+        if space_name in args.white_space_in:
+            return path.rstrip('/' + os.sep) + ' space'
+        return path
     args.workspace = 'work space' if 'workspace' in args.white_space_in else 'ws'
-    args.sourcespace = 'source space' if 'sourcespace' in args.white_space_in else 'src'
-    args.buildspace = 'build space' if 'buildspace' in args.white_space_in else 'build'
-    args.installspace = 'install space' if 'installspace' in args.white_space_in else 'install'
+    args.sourcespace = add_white_space_to_path(args.sourcespace, 'sourcespace')
+    args.buildspace = add_white_space_to_path(args.buildspace, 'buildspace')
+    args.installspace = add_white_space_to_path(args.installspace, 'installspace')
```

Comment:
How about simplifying the logic to something like this:

```
for name in ('sourcespace', 'buildspace', 'installspace'):
    if name in args.white_space_in:
        value = getattr(args, name)
        assert value.endswith('space')
        value = value[:-5] + ' ' + value[-5:]
        setattr(args, name, value)
```

Reply:
Wouldn't that assertion fail for the default values of source/build/install space? I was trying to make it so that neither white_space_in, nor the source/build/install-space arguments had precedence over one another.

So the following cases would still work:
'build' -> 'build space'
'/path/to/build' -> '/path/to/build space'
'/path/to/build/' -> '/path/to/build space'

Or on windows:
'C:\\path\\to\\build\\' -> 'C:\\path\\to\\build\\'
'C:/path/to/build/' -> 'C:/path/to/build space'

Reply:
The assertion was just an example. Feel free to use any other approach to determine the modified `value`. My suggestion was mostly about the overall logic.